documentation/Pure DP Mechanism.lyx

```diff
@@ -0,0 +1,652 @@
+ Let 
+\begin_inset Formula $\mathbb{F}_{p}$
+\end_inset
+
+ be field of prime order over which Prio3 operates.
+ Noise is sampled from the discrete Laplace distribution 
+\begin_inset Formula $\mathrm{Lap}_{\mathbb{Z}}\left(\nicefrac{GS_{f}}{\varepsilon}\right)$
+\end_inset
+
+,
+ projected into the field,
+ and added to each coordinate of aggregate share field element vectors.
+ Let 
+\begin_inset Formula $\pi_{\mathbb{F}_{p}}:\mathbb{Z}\rightarrow\mathbb{F}_{p}$
+\end_inset
+
+ and 
+\begin_inset Formula $\pi_{\mathbb{Z}}:\mathbb{F}_{p}\rightarrow\mathbb{Z}$
+\end_inset
+
+ be the natural projections between the integers and field elements,
+ where 
+\family roman
+\series medium
+\shape up
+\size normal
+\emph off
+\nospellcheck off
+\bar no
+\strikeout off
+\xout off
+\uuline off
+\uwave off
+\noun off
+\color none
+
+\begin_inset Formula $\pi_{\mathbb{Z}}$
+\end_inset
+
+
+\family default
+\series default
+\shape default
+\size default
+\emph default
+\nospellcheck default
+\bar default
+\strikeout default
+\xout default
+\uuline default
+\uwave default
+\noun default
+\color inherit
+ maps field elements to 
+\begin_inset Formula $\left[0,p\right)$
+\end_inset
+
+.
+ Let 
+\begin_inset Formula $q^{*}=f^{*}\left(x\right)\in\mathbb{F}_{p}^{d}$
+\end_inset
+
+ be the element-wise projections of 
+\begin_inset Formula $q$
+\end_inset
+
+ and 
+\begin_inset Formula $f$
+\end_inset
+
+ into the field using 
+\begin_inset Formula $\pi_{\mathbb{F}_{p}}$
+\end_inset
+
+.
+ The un-noised aggregate shares produced by Prio3 are secret shares of the query result,
+ 
+\begin_inset Formula $q^{*}=q^{\left(0\right)}+q^{\left(1\right)}$
+\end_inset
+
+.
+ Each aggregator samples noise from the discrete Laplace distribution and adds it to the un-noised aggregate shares,
+ and then sends the sum as their aggregate share to the collector.
+ If we pessimistically assume that only one honest aggregator out of the two aggregators is adding differential privacy noise,
+ then the mechanism produces 
+\begin_inset Formula $M\left(x\right)=q^{\left(0\right)}+q^{\left(1\right)}+\pi_{\mathbb{F}_{p}}\left(Z\right)=q^{*}+\pi_{\mathbb{F}_{p}}\left(Z\right)$
+\end_inset
+
+,
+ where 
+\begin_inset Formula $Z_{j}\leftarrow\mathrm{Lap}_{\mathbb{Z}}\left(\nicefrac{GS_{f}}{\varepsilon}\right)$
+\end_inset
+
+ is drawn independently for all 
+\begin_inset Formula $1\leq j\le d$
+\end_inset
+
+.
+\end_layout
+
+\begin_layout Theorem
+\begin_inset Formula $M\left(x\right)=\pi_{\mathbb{F}_{p}}\left(f\left(x\right)\right)+\pi_{\mathbb{F}_{p}}\left(Z\right),Z_{j}\leftarrow Lap_{\mathbb{Z}}\left(\nicefrac{GS_{f}}{\varepsilon}\right)$
+\end_inset
+
+ satisfies 
+\begin_inset Formula $\varepsilon$
+\end_inset
+
+-differential privacy.
+\end_layout
+
+\begin_layout Proof
+We will show Definition 1 holds for singleton events,
+ where 
+\begin_inset Formula $E$
+\end_inset
+
+ is a set of cardinality one,
+ then other events will follow by a union bound.
```

Comment:
I don't have an intuition for this step. Is the union bound over the power set of $\mathbb{Z}^d$?